x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/mapper/AggregateMetricFieldMapper.java

```diff
@@ -0,0 +1,340 @@
+    public AggregateMetricFieldType fieldType() {
+        return (AggregateMetricFieldType) super.fieldType();
+    }
+
+    @Override
+    protected String contentType() {
+        return fieldType.typeName();
+    }
+
+    @Override
+    protected AggregateMetricFieldMapper clone() {
+        return (AggregateMetricFieldMapper) super.clone();
+    }
+
+    @Override
+    protected void parseCreateField(ParseContext context, List<IndexableField> fields) throws IOException {
+        Object metricsAsObject;
+        if (context.externalValueSet()) {
+            metricsAsObject = context.externalValue();
+        } else {
+            metricsAsObject = context.parser().map();
+        }
+
+        if (metricsAsObject == null) {
+            metricsAsObject = fieldType().nullValue();
+        }
+
+        if (metricsAsObject == null) {
+            return;
+        }
+
+        Map<String, Object> parsedMetricsMap = new LinkedHashMap<>();
+        if (metricsAsObject instanceof Map) {
+            @SuppressWarnings("unchecked")
+            Map<String, Object> m = (Map<String, Object>) metricsAsObject;
+            for (Map.Entry<String, Object> e : m.entrySet()) {
+                String metricName = e.getKey();
+
+                if (metrics.value().contains(metricName) == false) {
+                    if (ignoreMalformed.value() == false) {
+                        throw new IllegalArgumentException("Aggregate metric [" + metricName +
+                            "] does not exist in the mapping of field [" + fieldType.name() + "]");
+                    } else {
+                        context.addIgnoredField(fieldType.name());
+                        continue;
+                    }
+                }
+
+                // Make sure that the value is a number. Probably this will change when
+                // new aggregate metric types are added (histogram, cardinality etc)
+                if (e.getValue() instanceof Number == false) {
+                    if (ignoreMalformed.value() == false) {
+                        throw new IllegalArgumentException("Aggregate metric [" + metricName +
+                            "] of field [" + fieldType.name() + "] must be a number");
+                    } else {
+                        context.addIgnoredField(fieldType.name());
+                        continue;
+                    }
+                }
+
+                // Make sure that value_count is not a negative value
+                if (Metrics.value_count.toString().equals(metricName) == true) {
+                    if (e.getValue() instanceof Integer == false) {
+                        if (ignoreMalformed.value() == false) {
+                            throw new IllegalArgumentException("Aggregate metric [" + metricName +
+                                "] of field [" + fieldType.name() + "] must be an integer number");
+                        } else {
+                            context.addIgnoredField(fieldType.name());
```

Comment:
I don't think we should just ignore and go to the next entry of the map if ignore malformed is true. My take is that we should not generate the doc value and ignore this field.